src/types.rs

```diff
@@ -32,6 +34,38 @@ pub struct Amount {
     pub micro_ccd: u64,
 }
 
+#[cfg(all(feature = "std", target_arch = "wasm32"))]
+use std::sync::atomic::AtomicU8;
+#[cfg(all(feature = "std", target_arch = "wasm32"))]
+static ADDRESS_COUNTER: AtomicU8 = AtomicU8::new(0);
+#[cfg(all(feature = "std", target_arch = "wasm32"))]
+use getrandom::register_custom_getrandom;
+#[cfg(all(feature = "std", target_arch = "wasm32"))]
+fn custom_random(dest: &mut [u8]) -> Result<(), getrandom::Error> {
+    for byte in dest.iter_mut() {
+        *byte = ADDRESS_COUNTER.load(std::sync::atomic::Ordering::SeqCst);
+        ADDRESS_COUNTER.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+    }
+    Ok(())
+}
+#[cfg(all(feature = "std", target_arch = "wasm32"))]
+register_custom_getrandom!(custom_random);
+
+#[cfg(feature = "std")]
+impl quickcheck::Arbitrary for Amount {
+    fn arbitrary(g: &mut Gen) -> Amount {
+        Amount {
+            micro_ccd: quickcheck::Arbitrary::arbitrary(g),
+        }
+    }
+
+    fn shrink(&self) -> Box<dyn Iterator<Item = Self>> {
+        Box::new(quickcheck::Arbitrary::shrink(&self.micro_ccd).map(|mccd| Amount {
+            micro_ccd: mccd,
+        }))
+    }
+}
```

Comment:
This looks like something we could derive automatically, does it not? https://docs.rs/arbitrary/1.1.0/arbitrary/trait.Arbitrary.html#deriving-arbitrary

Is there a reason I'm not seeing for doing it manually?

Reply:
Turns out [arbitrary::Arbitrary](https://docs.rs/arbitrary/1.1.0/arbitrary/trait.Arbitrary.html#deriving-arbitrary) is not the same as [quickcheck::Arbitrary](https://docs.rs/quickcheck/latest/quickcheck/trait.Arbitrary.html). The one from quickcheck doesn't have built-in `derive`. [This crate](https://github.com/panicbit/quickcheck_derive.) looks outdated and I couldn't make it work.

Also, the PR adding this feature was [abandoned](https://github.com/BurntSushi/quickcheck/pull/222#issuecomment-573267560). So, I guess we just keep these instances.